plugins/uploadimage/plugin.js

```diff
@@ -6,6 +6,29 @@
 'use strict';
 
 ( function() {
+	var uniqueNameCounter = 0,
+		// Black rectangle which is shown before image is loaded.
+		loadingImage = 'data:image/gif;base64,R0lGODlhDgAOAIAAAAAAAP///yH5BAAAAAAALAAAAAAOAA4AAAIMhI+py+0Po5y02qsKADs=';
+
+	// Returns number as a string. If a number has 1 digit only it returns it prefixed with an extra 0.
+	function padNumber( input ) {
+		if ( input <= 9 ) {
+			input = '0' + input;
+		}
+
+		return String( input );
+	}
+
+	// Returns an unique image file name.
+	function getUniqueImageFileName( type ) {
+		var date = new Date(),
+			dateParts = [ date.getFullYear(), date.getMonth() + 1, date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds() ];
```

Comment:
Couldn't we use simply timestamp here, returned from `Date.now()`/`Date.prototype.getTime()`?

Reply:
We want to give it _some_ human readability, at the same time not making the name too long.